saleor/graphql/product/resolvers.py

```diff
@@ -41,5 +25,25 @@ def resolve_attributes(category_id, info):
     return queryset.distinct()
 
 
+def resolve_categories(info, level=None):
+    qs = models.Category.objects.all()
+    if level is not None:
+        qs = qs.filter(level=level)
+    return qs.distinct()
+
+
+def resolve_collections(info):
+    return models.Collection.objects.all()
```

Comment:
We need to make sure that we exclude unpublished collections for unauthorized requests.